dev/com.ibm.ws.cdi.2.0.weld/src/com/ibm/ws/cdi/impl/weld/AbstractProxyServices.java

```diff
@@ -155,6 +160,19 @@ public ClassLoader run() {
                 }
                 Class<?> clazz = (Class) method.invoke(loader, args);
                 return clazz;
+
+             } catch (LinkageError e) {
+                int tries = 10;
+                while (tries > 0) {
+                    Class<?> clazz = loadClass​(className, loader);
+                    if (clazz != null) {
+                        return clazz;
+                    }
+                    Thread.sleep(1000);
```

Comment:
definitely put comments here ... I don't like a Thread.sleep really

Reply:
Why does it need the sleep anyway?